Repository/PhpcrOdmRepository.php

```diff
@@ -116,4 +128,62 @@ protected function buildCollection(array $documents)
 
         return $collection;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function add($path, $resource)
+    {
+        Assert::notEq('', trim($path, '/'), 'You cannot create a root node.');
+        Assert::startsWith($path, '/', 'Target path "%s" must be absolute.');
+
+        $resolvedPath = $this->resolvePath($path);
+        $parentPath = PathHelper::getParentPath($resolvedPath);
+        $parentDocument = $this->getManager()->find(null, $parentPath);
+
+        if (null === $parentDocument) {
+            throw new InvalidArgumentException(sprintf('Cannot locate parent document at "%s"', $parentPath));
+        }
+
+        /** @var PhpcrOdmResource[] $resources */
+        $resources = $resource instanceof IteratorAggregate ? $resource : new ArrayResourceCollection([$resource]);
+        Assert::isInstanceOf($resources, ResourceCollection::class, 'The list should be of instance "ResourceCollection".');
+
+        foreach ($resources as $resource) {
+            Assert::isInstanceOf($resource, CmfResource::class, 'The resource needs to of instance "CmfResource".');
+            Assert::notNull($resource->getName(), 'The resource needs a name for the creation.');
+
+            $document = $resource->getPayload();
+            $document->setName($resource->getName());
+            if ($document instanceof HierarchyInterface) {
```

Comment:
Documents do not require a "name" method, they also do not need to implement the `HierearchyInterface` in order to have a parent, nor would they necessarily have either a "parent" or a "name" mapping as they could be using an "ID/path" ID strategy (where you would set the path directly).

In the Syluus resource bundle I force the user to use the `name` and `parent` mappings, but that could be more flexible.

I am not even sure it makes sense to be able to create documents here at all - you cannot gaurantee that a document doesn't have required fields which cannot be set using this API.

/cc @dbu @WouterJ